kernel/kernel-defaults/src/main/java/io/delta/kernel/defaults/internal/expressions/DefaultExpressionEvaluator.java

```diff
@@ -589,6 +605,56 @@ ColumnVector visitCoalesce(ScalarExpression coalesce) {
           });
     }
 
+    @Override
+    ColumnVector visitAdd(ScalarExpression add) {
+      List<ColumnVector> childResults =
+          add.getChildren().stream().map(this::visit).collect(toList());
+      if (childResults.size() != 2) {
+        throw unsupportedExpressionException(
+            add, "ADD requires exactly two arguments: left and right operands");
+      }
+
+      return DefaultExpressionUtils.arithmeticVector(
+          childResults.get(0),
+          childResults.get(1),
+          new ArithmeticOperator() {
```

Comment:
This implementation is good enough for now, add a comment though to mention that:
- it only supports adding operands of the same type
- it doesn't check for overflow and uses java semantics: integer values will wrap around when overflowing